processing/src/main/java/org/apache/druid/segment/join/Joinables.java

```diff
@@ -114,4 +118,37 @@ public static boolean isPrefixedBy(final String columnName, final String prefix)
       );
     }).collect(Collectors.toList());
   }
+
+  public static void checkPreJoinableClausesForDuplicatesAndShadowing(
+      final List<PreJoinableClause> preJoinableClauses
+  )
+  {
+    List<String> prefixes = new ArrayList<>();
+    for (PreJoinableClause clause : preJoinableClauses) {
+      prefixes.add(clause.getPrefix());
+    }
+
+    checkPrefixesForDuplicatesAndShadowing(prefixes);
+  }
+
+  public static void checkPrefixesForDuplicatesAndShadowing(
```

Comment:
```suggestion
  private static void checkPrefixesForDuplicatesAndShadowing(
```

Reply:
I kept this public, it's used in tests and I think it could be useful as a general utility method